cli/src/commands/install.js

```diff
@@ -63,19 +62,114 @@ export async function run(args: Args): Promise<number> {
   }
   const flowVersion = stringToVersion(flowVersionStr);
 
-  const term = args._[1];
+  // Find the project root
+  const projectRoot = await findFlowProjectRoot();
+  if (projectRoot === null) {
+    return failWithMessage(
+      `\nERROR: Unable to find a flow project in the currend dir or any of ` +
+      `it's parents!\nPlease run this command from within a Flow project.`
+    );
+  }
 
-  const matches = term.match(/([^@]*)@?(.*)?/);
-  const defName = (matches && matches[1]);
-  const defVersion = (matches && matches[2]) || 'auto';
+  // generate a map { dependency: version } for libs to install
+  // from command line or package.json.
+  let depsMap: DepsMap = {};  
+  if(args._[1]) {
+    const term = args._[1];
+    const matches = term.match(/([^@]*)@?(.*)?/);
+    if(!matches) {
+      console.error(      
+        "Please specify a package name of the format `PackageFoo` or " +
+        "PackageFoo@0.2.2"
+      );
+    }
+    
+    const defName = (matches && matches[1]);
+    const defVersion = (matches && matches[2]) || 'auto';
+    if(defName) {
+      depsMap[defName] = defVersion;
+    }
+  } else {
+    depsMap = await getInstalledPackageDependencies(process.cwd());
+  }
 
-  if (!defName) {
+  if (!Object.keys(depsMap).length === 0) {
     return failWithMessage(
-      "Please specify a package name of the format `PackageFoo` or " +
-      "PackageFoo@0.2.2"
+      "No package dependencies were found in package.json or specified on the command line."
     );
   }
 
+  // Get a list of defs to install
+  const defs: [LibDef] = [];
+  for(let dep in depsMap) {
+    console.log(`Found npm dependency: ${dep} v${depsMap[dep]}`);
+    let def = await findLibDef(
+      dep, 
+      depsMap[dep],
```

Comment:
This is going to be where things get tricky: `package.json` holds version ranges which are a super set of version strings themselves...and we don't really handle them yet.

Moreover, flow-typed versions libdefs with ranges as well -- so this means we're going to have to compare the range specified for a dependency in `package.json` with all of the possible ranges available for a given package in flow-typed and choose the "best" one.

Making the choice basically amounts to picking the version of the package that is both the lowest-compatible version of the range in package.json as well as the lowest-compatible version in flow-typed.

The easiest way to do this would be to have a list of all the versions for a given package -- but the caveat there is that would require calling out to npm to query the list of versions. A simpler way might be to find the lowest-possible version for each range (package.json range, and all the flow-typed ranges) and choose the lowest matching range in flow-typed.